File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/constover_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'constover'

error = 0

p = Constover.test("test")
if p != "test"
  puts "test failed!"
  error = 1
end

p = Constover.test_pconst("test")
if p != "test_pconst"
  puts "test_pconst failed!"
  error = 1
end
    
f = Constover::Foo.new
p = f.test("test")
if p != "test"
  print "member-test failed!"
  error = 1
end

p = f.test_pconst("test")
if p != "test_pconst"
  print "member-test_pconst failed!"
  error = 1
end

p = f.test_constm("test")
if p != "test_constmethod"
  print "member-test_constm failed!"
  error = 1
end

p = f.test_pconstm("test")
if p != "test_pconstmethod"
  print "member-test_pconstm failed!"
  error = 1
end
    
exit(error)

File: support_tools/swigwin-3.0.12/Examples/test-suite/go/constover_runme.go
package main

import (
	"./constover"
	"fmt"
	"os"
)

func main() {
	error := 0

	p := constover.Test("test")
	if p != "test" {
		fmt.Println("test failed!")
		error = 1
	}

	p = constover.Test_pconst("test")
	if p != "test_pconst" {
		fmt.Println("test_pconst failed!")
		error = 1
	}

	f := constover.NewFoo()
	p = f.Test("test")
	if p != "test" {
		fmt.Println("member-test failed!")
		error = 1
	}

	p = f.Test_pconst("test")
	if p != "test_pconst" {
		fmt.Println("member-test_pconst failed!")
		error = 1
	}

	p = f.Test_constm("test")
	if p != "test_constmethod" {
		fmt.Println("member-test_constm failed!")
		error = 1
	}

	p = f.Test_pconstm("test")
	if p != "test_pconstmethod" {
		fmt.Println("member-test_pconstm failed!")
		error = 1
	}

	os.Exit(error)
}
